pennylane/devices/default_mixed.py

```diff
@@ -118,13 +119,52 @@ def _create_basis_state(self, index):
         rho = self._asarray(rho, dtype=self.C_DTYPE)
         return self._reshape(rho, [2] * (2 * self.num_wires))
 
+    @classmethod
+    def capabilities(cls):
+        capabilities = super().capabilities().copy()
+        capabilities.update(
+            returns_state=True,
+        )
+        return capabilities
+
     @property
     def state(self):
         """Returns the state density matrix of the circuit prior to measurement"""
         dim = 2 ** self.num_wires
         # User obtains state as a matrix
         return self._reshape(self._pre_rotated_state, (dim, dim))
 
+    def density_matrix(self, wires=None):
```

Comment:
You are right it should not be equal to `None` because the `density_matrix` is called if wires were given `if wires`. This was done in order to differentiate the function `state() = MeasurementProcess(State)` and `density_matrix(wires) = MeasurementProcess(State, wires)`. As we are dealing with quantum system, trace out all subsystems will always return 1 and therefore I don't think it is useful to implement it. Consequently I'll just remove the default argument.